app-example/src/main/java/com/badoo/ribs/example/photo_details/feature/PhotoDetailsFeature.kt

```diff
@@ -0,0 +1,126 @@
+package com.badoo.ribs.example.photo_details.feature
+
+import com.badoo.mvicore.element.Actor
+import com.badoo.mvicore.element.Bootstrapper
+import com.badoo.mvicore.element.NewsPublisher
+import com.badoo.mvicore.element.Reducer
+import com.badoo.mvicore.feature.ActorReducerFeature
+import com.badoo.ribs.example.auth.AuthDataSource
+import com.badoo.ribs.example.auth.AuthState.Authenticated
+import com.badoo.ribs.example.extensions.toObservable
+import com.badoo.ribs.example.network.model.Photo
+import com.badoo.ribs.example.photo_details.PhotoDetailsDataSource
+import com.badoo.ribs.example.photo_details.feature.PhotoDetailsFeature.Effect
+import com.badoo.ribs.example.photo_details.feature.PhotoDetailsFeature.News
+import com.badoo.ribs.example.photo_details.feature.PhotoDetailsFeature.State
+import com.badoo.ribs.example.photo_details.feature.PhotoDetailsFeature.Wish
+import io.reactivex.Observable
+
+internal class PhotoDetailsFeature(
+    photoId: String,
+    photoDetailsDataSource: PhotoDetailsDataSource,
+    authDataSource: AuthDataSource
+) : ActorReducerFeature<Wish, Effect, State, News>(
+    initialState = State.Loading,
+    bootstrapper = BootStrapperImpl(),
+    actor = ActorImpl(photoDetailsDataSource, photoId, authDataSource),
+    reducer = ReducerImpl(),
+    newsPublisher = NewsPublisherImpl(authDataSource)
+) {
+
+
+    sealed class State {
+        object Loading : State()
+        data class Loaded(val detail: Photo) : State()
+    }
+
+    sealed class Wish {
+        object LoadPhoto : Wish()
+        object LikePhoto : Wish()
+    }
+
+    sealed class Effect {
+        object LoadingStarted : Effect()
+        data class PhotoLoaded(val detail: Photo) : Effect()
+        object LikeSent : Effect()
+        object UnlikeSent : Effect()
+        object LikeNotSent : Effect()
+    }
+
+    sealed class News {
+        object ShowLogin : News()
```

Comment:
This should be named report-style, not command-style. Let the handler decide what to do next, only report what happened at this point.